File: 260-Single-Number-III/260-Single-Number-III.java
class Solution {
    public int[] singleNumber(int[] nums) {
        if(nums.length == 2) return nums;
        HashSet<Integer> set = new HashSet<>();
        for(int i = 0; i < nums.length; i++){
            if(set.contains(nums[i])){
                set.remove(nums[i]);
            } else {
                set.add(nums[i]);
            }
        }
        int[] ans = new int[2];
        int pointer = 0;
        for(Integer i : set){
            ans[pointer] = i;
            pointer++;
        }
        return ans;

    }
}

File: 260-Single-Number-III/260-Single-Number-III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        if len(nums) == 2:
            return nums
        ans = []
        for i in nums:
            if i not in ans:
                ans.append(i)
            else:
                ans.remove(i)
        return ans